app/controlplane/internal/usercontext/attjwtmiddleware/attmiddleware.go

```diff
@@ -0,0 +1,269 @@
+		WithVerifyAudienceFunc(func(token *jwt.Token) bool {
+			claims, ok := token.Claims.(*robotaccount.CustomClaims)
+			if !ok {
+				return false
+			}
+			for _, aud := range []string{robotaccount.Audience, robotaccount.DeprecatedAudience} {
+				if claims.VerifyAudience(aud, true) {
+					return true
+				}
+			}
+			return false
+		}),
+		WithSigningMethod(robotaccount.SigningMethod),
+		WithKeyFunc(func(_ *jwt.Token) (interface{}, error) {
+			// TODO: add support to multiple signing methods and keys
+			return []byte(signingSecret), nil
+		}),
+	)
+}
+
+// NewAttestationAPITokenProvider return the configuration to validate and verify token issued for API Tokens
+func NewAttestationAPITokenProvider(signingSecret string) JWTOption {
+	return WithTokenProvider(
+		APITokenProviderKey,
+		WithVerifyAudienceFunc(func(token *jwt.Token) bool {
+			claims, ok := token.Claims.(jwt.MapClaims)
+			if !ok {
+				return false
+			}
+			for _, aud := range []string{apitoken.Audience} {
+				if claims.VerifyAudience(aud, true) {
+					return true
+				}
+			}
+			return false
+		}),
+		WithSigningMethod(user.SigningMethod),
+		WithKeyFunc(func(_ *jwt.Token) (interface{}, error) {
+			return []byte(signingSecret), nil
+		}),
+	)
+}
+
+type JWTAuthContext struct {
+	Claims      jwt.Claims
+	ProviderKey string
+}
+
+type authzContextKey struct{}
+
+type JWTOption func(*options)
+type TokenProviderOption func(*providerOption)
+
+type VerifyAudienceFunc func(*jwt.Token) bool
+
+type providerOption struct {
+	providerKey        string
+	signingMethod      jwt.SigningMethod
+	keyFunc            jwt.Keyfunc
+	claims             func() jwt.Claims
+	verifyAudienceFunc VerifyAudienceFunc
+}
+
+func WithSigningMethod(method jwt.SigningMethod) TokenProviderOption {
+	return func(o *providerOption) {
+		o.signingMethod = method
+	}
+}
+
+func WithClaims(f func() jwt.Claims) TokenProviderOption {
+	return func(o *providerOption) {
+		o.claims = f
+	}
+}
+
+func WithKeyFunc(keyFunc jwt.Keyfunc) TokenProviderOption {
+	return func(o *providerOption) {
+		o.keyFunc = keyFunc
+	}
+}
+
+func WithVerifyAudienceFunc(f VerifyAudienceFunc) TokenProviderOption {
+	return func(o *providerOption) {
+		o.verifyAudienceFunc = f
+	}
+}
+
+type options struct {
+	tokenProviders []providerOption
+}
+
+func WithTokenProvider(providerKey string, opts ...TokenProviderOption) JWTOption {
+	op := &providerOption{
+		providerKey: providerKey,
+	}
+	for _, opt := range opts {
+		opt(op)
+	}
+	return func(o *options) {
+		o.tokenProviders = append(o.tokenProviders, *op)
+	}
+}
+
+// WithAttestationTokenMiddleware creates a custom JWT middleware that configured with different token providers
+// tries to run all validations from an incoming token. If you of the providers matches the expected audience
+// it gets parsed and sent down to the next middleware
+func WithAttestationTokenMiddleware(opts ...JWTOption) middleware.Middleware {
```

Comment:
We might want to indicate in the name that this is a multi-selector kind of middleware.

Also, the package is already called `attmiddleware` so we do not need to have the full name in the function.

What about `JWTMulti` or similar so we express two things, this is about handling the `jwt` (that's not shown anywhere anymore), and that's a multi selector.